### ISSUE TITLE
Enable RSpec/NamedSubject cop

### ISSUE THREAD

**Issue by makesime** [petalmd/rubocop-petal#67]
Following the Rspec Guideline, we should use a named `subject` when possible, therefore I'm suggesting that we enable the `RSpec/NamedSubject` cop to enforce the use of a named `subject`.

Cop: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecnamedsubject

Reference:

> Use named subject when possible. Only use anonymous subject declaration when you don’t reference it in any tests, e.g. when is_expected is used.
https://rspec.rubystyle.guide/#use-subject

**Comment by Bhacaz** [comment]
I agree only if we change the config of https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecimplicitsubject

I think it loose all the interest of `subject` if we always have to reference to it with a name it is just like using `let`.

Example

```ruby
# Now
subject { create :user }
it { expect(subject).to be_valid? }

# With cop
subject(:user) { create :user }
it { expect(user).to be_valid? }

# Is the same

let(:user) { create :user }
it { expect(user).to be_valid? }
```

If we change `RSpec/ImplicitSubject` to `require_implicit` AND `RSpec/NamedSubject` to `named_only` I think it will the best of both. So a name will be required only when "subject" is called AND you cannot call directly subject.

```ruby
# Good subject implicitly used
subject { create :user }
it { is_expected.to be_valid }

# Bad, not named and called
subject { create :user }
it { expect(subject).to be_valid }

# Good
subject(:user) { create :user }
it do
 is_expected.to be_valid
 expect(user.email).to be_present
end
```

Thoughts?

**Comment by makesime** [comment]
I agree with your suggestion ✅ 

For the part:
> I think it loose all the interest of subject if we always have to reference to it with a name it is just like using let.

In my mind, both are meant to reduce unnecessary duplication anyway (so the same means). Using `subject` instead of `let`, even if they are the same should help to identify what is the action or the "subject" of the test, which would be "less" explicit with a `let`.
When reading definitions, we can see that they're pretty much the same so the "purpose" of the `subject` would not be loss even with a named `subject`:

> [Subject](https://rspec.rubystyle.guide/#subject)
When several tests relate to the same subject, use subject to reduce repetition.

> Let
Use let and let! for data that is used across several examples in an example group
Use let to reduce duplication throughout an entire spec file.

With that being said, I agree with your suggestion, I just think that a named `subject` still has an interest even if "it's the same" as a `let`.
